packages/cpp/src/browser/cpp-commands.ts

```diff
@@ -59,4 +99,30 @@ export class CppCommandContribution implements CommandContribution {
             });
         });
     }
+
+    protected changeConfiguration(): void {
+        this.workspaceService.tryRoot.then(async resolvedRoot => {
+            const root = resolvedRoot || await this.fileSystem.getCurrentUserHome();
```

Comment:
Not a typescript expert, but I think having many awaits means waiting synchronously. Should those places use .then instead? Haven' tried though...

Reply:
await doesn't mean sync.. await will basically make this execution stack return and async wait for the right condition to continue
It would look better like
```
root = tryRoot;

if root === undefined
  root = await.this.filesystem.getCurrentUserHome()
```